### PR TITLE
Merge v1.0.2_spec: update to new ApplicationPattern release 2.1.2 (outdated)

### PR DESCRIPTION
**Pull request by kmohr-soprasteria**
The specification has been updated to use the new appPattern v2.1.2.
Alongside it, also the MWDI release version has been updated.

See issue #240

### CODE REVIEW

File: spec/MacAddressTableRecorder.yaml [hunk 1]
```diff
@@ -2757,27 +2902,27 @@ paths:
                   type: string
                   description: >
                     'Name of application that shall record the OaM requests
-                    find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-http-c-ol-2-0-1-000/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/application-name]'
+                    find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-http-c-ol-2-1-2-000/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/application-name]'
```
Comment (PrathibaJee) [inline spec/MacAddressTableRecorder.yaml:2905]: As per the decision https://github.com/openBackhaul/ApplicationPattern/issues/714#issuecomment-1594314255, "find" needs to modified to "find or update, but not at OldRelease or NewRelease,"

File: spec/MacAddressTableRecorder.yaml [hunk 2]
```diff
@@ -2757,27 +2902,27 @@ paths:
...
                 oam-log-application-release-number:
                   type: string
                   pattern: '^([0-9]{1,2})\.([0-9]{1,2})\.([0-9]{1,2})$'
                   description: >
                     'Release of application that shall record the OaM request
-                    update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-http-c-ol-2-0-1-000/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/release-number]'
+                    update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-http-c-ol-2-1-2-000/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/release-number]'
```
Comment (PrathibaJee) [inline spec/MacAddressTableRecorder.yaml:2911]: As per the decision https://github.com/openBackhaul/ApplicationPattern/issues/714#issuecomment-1594314255, "find" needs to modified to "find or update, but not at OldRelease or NewRelease,"

File: spec/MacAddressTableRecorder.yaml [hunk 3]
```diff
@@ -2794,84 +2939,84 @@ paths:
...
-        '401':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '403':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '404':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '500':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
+                'Life cycle state of the consumed service
+                find in [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-op-s-bm-003/layer-protocol=0/operation-server-interface-1-0:operation-server-interface-pac/operation-server-interface-configuration/life-cycle-state]'
```
Comment (PrathibaJee) [inline spec/MacAddressTableRecorder.yaml:2997]: "find in" shall be changed to "from"

File: spec/MacAddressTableRecorder.yaml [hunk 4]
```diff
@@ -2997,78 +3142,78 @@ paths:
...
-        '401':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '403':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '404':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '500':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
+                'Life cycle state of the consumed service
+                find in [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-op-s-bm-004/layer-protocol=0/operation-server-interface-1-0:operation-server-interface-pac/operation-server-interface-configuration/life-cycle-state]'
```
Comment (PrathibaJee) [inline spec/MacAddressTableRecorder.yaml:3194]: "find in" shall be changed to "from"

File: spec/MacAddressTableRecorder.yaml [hunk 5]
```diff
@@ -2997,78 +3142,78 @@ paths:
...
+        "403":
+          $ref: "#/components/responses/responseForErroredServiceRequests"
+        "404":
+          $ref: "#/components/responses/responseForErroredServiceRequests"
+        "500":
+          $ref: "#/components/responses/responseForErroredServiceRequests"
         default:
-          $ref: '#/components/responses/responseForErroredServiceRequests'
+          $ref: "#/components/responses/responseForErroredServiceRequests"
   /v1/inquire-oam-request-approvals:
```
Comment (PrathibaJee) [inline spec/MacAddressTableRecorder.yaml:3207]: The following description shall be added to this path , 
```
description: >
      'DEPRECATED
      Service shall be deleted with AP v3.0.0
      Starting from AP v2.1.0, /v1/inquire-basic-auth-approvals shall be applied as a backward compatible update'
```

File: spec/MacAddressTableRecorder.yaml [hunk 6]
```diff
@@ -3091,27 +3236,27 @@ paths:
                   type: string
                   description: >
                     'Name of application that shall approve the OaM requests
-                    find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-http-c-aa-2-0-1-000/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/application-name]'
+                    find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-http-c-aa-2-1-2-000/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/application-name]'
```
Comment (PrathibaJee) [inline spec/MacAddressTableRecorder.yaml:3239]: As per the decision https://github.com/openBackhaul/ApplicationPattern/issues/714#issuecomment-1594314255, "find" needs to modified to "find or update, but not at OldRelease or NewRelease,"

File: spec/MacAddressTableRecorder.yaml [hunk 7]
```diff
@@ -3091,27 +3236,27 @@ paths:
...
                 oam-approval-application-release-number:
                   type: string
                   pattern: '^([0-9]{1,2})\.([0-9]{1,2})\.([0-9]{1,2})$'
                   description: >
                     'Release of application that shall approve the OaM requests
-                    update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-http-c-aa-2-0-1-000/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/release-number]'
+                    update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-http-c-aa-2-1-2-000/layer-protocol=0/http-client-interface-1-0:http-client-interface-pac/http-client-interface-configuration/release-number]'
```
Comment (PrathibaJee) [inline spec/MacAddressTableRecorder.yaml:3245]: As per the decision https://github.com/openBackhaul/ApplicationPattern/issues/714#issuecomment-1594314255, "find" needs to modified to "find or update, but not at OldRelease or NewRelease,"

File: spec/MacAddressTableRecorder.yaml [hunk 8]
```diff
@@ -3128,84 +3273,84 @@ paths:
...
-        '401':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '403':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '404':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '500':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
+                'Life cycle state of the consumed service
+                find in [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-op-s-bm-005/layer-protocol=0/operation-server-interface-1-0:operation-server-interface-pac/operation-server-interface-configuration/life-cycle-state]'
```
Comment (PrathibaJee) [inline spec/MacAddressTableRecorder.yaml:3331]: "find in" shall be changed to "from"

**Review by PrathibaJee — CHANGES_REQUESTED** [review]
Along with the other review comments , in the OAS from line no 3345, please refer AP2.1.2 in the latest TAC application and update accordingly.

File: spec/MacAddressTableRecorder.yaml [hunk 9]
```diff
@@ -1968,39 +2113,39 @@ paths:
                   type: string
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:2113]: fixed in branch for for #243

File: spec/MacAddressTableRecorder.yaml [hunk 10]
```diff
@@ -2055,91 +2200,91 @@ paths:
...
-        '401':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '403':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '404':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '500':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
+                'Life cycle state of the consumed service
+                find in [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-op-s-bm-001/layer-protocol=0/operation-server-interface-1-0:operation-server-interface-pac/operation-server-interface-configuration/life-cycle-state]'
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:2265]: changed with branch for issue #243

File: spec/MacAddressTableRecorder.yaml [hunk 11]
```diff
@@ -1652,23 +1664,23 @@ paths:
...
                                 'If provided, own IPv4 address, which has to be documented (but not be directly used for formulating the callback) into the TcpServer with
-                                find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol] == http
-                                update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-address/ipv-4-address]'
+                                find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol] == http
+                                update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-address/ipv-4-address]'
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:1668]: fixed in branch for issue 243

File: spec/MacAddressTableRecorder.yaml [hunk 12]
```diff
@@ -1652,23 +1664,23 @@ paths:
...
                         domain-name:
                           type: string
                           pattern: '^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$'
                           description: >
                             'If provided, own domain name, which has to be documented (but not be directly used for formulating the callback) into the TcpServer with
-                            find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol] == http
-                            update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-address/domain-name]'
+                            find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol] == http
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:1674]: see above

File: spec/MacAddressTableRecorder.yaml [hunk 13]
```diff
@@ -1652,23 +1664,23 @@ paths:
...
+                            update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-address/domain-name]'
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:1675]: see above

File: spec/MacAddressTableRecorder.yaml [hunk 14]
```diff
@@ -1652,23 +1664,23 @@ paths:
...
                     http-port:
                       type: integer
                       minimum: 0
                       maximum: 65535
                       description: >
                         'If provided, own TCP port, which has to be documented (but not be directly used for formulating the callback) into the TcpServer with
-                        find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol] == http
-                        update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-port]'
+                        find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol] == http
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:1682]: see above

File: spec/MacAddressTableRecorder.yaml [hunk 15]
```diff
@@ -1652,23 +1664,23 @@ paths:
...
+                        update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-port]'
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:1683]: s.a.

File: spec/MacAddressTableRecorder.yaml [hunk 16]
```diff
@@ -1652,23 +1664,23 @@ paths:
...
                     https-address:
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:1684]: fixed, issue 243

File: spec/MacAddressTableRecorder.yaml [hunk 17]
```diff
@@ -1716,68 +1728,70 @@ paths:
...
-        '401':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '403':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '404':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '500':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
+                'Life cycle state of the consumed service
+                find in [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-op-s-bm-000/layer-protocol=0/operation-server-interface-1-0:operation-server-interface-pac/operation-server-interface-configuration/life-cycle-state]'
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:1777]: fixed

File: spec/MacAddressTableRecorder.yaml [hunk 18]
```diff
@@ -1652,23 +1664,23 @@ paths:
                               pattern: '^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'
                               description: >
                                 'If provided, own IPv4 address, which has to be documented (but not be directly used for formulating the callback) into the TcpServer with
-                                find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol] == http
-                                update [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-address/ipv-4-address]'
+                                find [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-tcp-s-*/layer-protocol=0/tcp-server-interface-1-0:tcp-server-interface-pac/tcp-server-interface-configuration/local-protocol] == http
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:1667]: fixed

File: spec/MacAddressTableRecorder.yaml (hunk 4 above)
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:3194]: fixed

File: spec/MacAddressTableRecorder.yaml (hunk 5 above)
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:3207]: fixed

File: spec/MacAddressTableRecorder.yaml [hunk 19]
```diff
@@ -2543,84 +2688,84 @@ paths:
...
-        '401':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '403':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '404':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
-        '500':
-          $ref: '#/components/responses/responseForErroredServiceRequests'
+                'Life cycle state of the consumed service
+                find in [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-op-s-bm-002/layer-protocol=0/operation-server-interface-1-0:operation-server-interface-pac/operation-server-interface-configuration/life-cycle-state]'
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:2746]: fixed, branch for 243

File: spec/MacAddressTableRecorder.yaml [hunk 20]
```diff
@@ -2543,84 +2688,84 @@ paths:
...
-                - $ref: '#/components/parameters/user'
-                - $ref: '#/components/parameters/originator'
-                - $ref: '#/components/parameters/x-correlator'
-                - $ref: '#/components/parameters/trace-indicator'
-                - $ref: '#/components/parameters/customer-journey'
+                - $ref: "#/components/parameters/user"
+                - $ref: "#/components/parameters/originator"
+                - $ref: "#/components/parameters/x-correlator"
+                - $ref: "#/components/parameters/trace-indicator"
+                - $ref: "#/components/parameters/customer-journey"
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:2768]: fixed

File: spec/MacAddressTableRecorder.yaml [hunk 21]
```diff
@@ -2639,102 +2784,102 @@ paths:
...
                         stringified-response:
                           type: string
-                          description: 'Stringified response sent from [application-name] to [originator]; only to be provided, if [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-0-op-c-bs-eatl-2-0-1-000/layer-protocol=0/operation-client-interface-1-0:operation-client-interface-pac/operation-client-interface-configuration/detailed-logging-is-on] == true'
+                          description: "Stringified response sent from [application-name] to [originator]; only to be provided, if [/core-model-1-4:control-construct/logical-termination-point=matr-1-0-2-op-c-bs-eatl-2-1-2-000/layer-protocol=0/operation-client-interface-1-0:operation-client-interface-pac/operation-client-interface-configuration/detailed-logging-is-on] == true"
```
[the remark this answers is not in the document]
Comment (kmohr-soprasteria) [reply spec/MacAddressTableRecorder.yaml:2817]: added

**Comment by kmohr-soprasteria** [comment]
issues were resolved in a new branch on develop (see #243); PR will be reopened

**Pull request closed without merging**